common/predictive-text/unit_tests/headless/default-word-breaker.js

```diff
@@ -19,4 +20,63 @@ describe('The default word breaker', function () {
       "let's", 'eat', 'phở', '!', '🥣'
     ]);
   });
+
+  // The following tests are performed with model integration as an internal
+  // test for the wordbreaking API.
+  it('recognizes a word at end of lefthand context', function () {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // Standard case - wordbreaking at the end of a word.
+    var context = { 
+      left: 'The quick brown fox jumped', startOfBuffer: true,
+      right: ' over the lazy dog.', endOfBuffer: true
+    };
+
+    var broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, 'jumped');
+  });
+
+  it('returns text for a word in-progress', function() {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // Standard case - midword (xylophone) call
+    var context = { 
+      left: 'xyl', startOfBuffer: true,
+      right: '', endOfBuffer: true
+    };
+
+    var broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, 'xyl');
+  });
+    
+  it('returns empty string when called without word text', function() {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // Wordbreaking on a empty space => no word.
+    context = { 
+      left: 'The quick brown fox jumped ', startOfBuffer: true,
+      right: 'over the lazy dog.', endOfBuffer: true
+    };
+
+    broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, '');
+  });
+
+  it.skip('correctly breaks a word when the caret is placed within it', function() {
+    var model = new TrieModel(jsonFixture('tries/english-1000'));
+
+    // A limitation of the current implementation; we should fix this before release.
+    // Then again, when typing this is probably fine; just not when not typing.
+    context = { 
+      left: 'The quick brown fox jum', startOfBuffer: true,
+      right: 'ped over the lazy dog.', endOfBuffer: true
+    };
+
+    broken = model.wordbreak(context);
+
+    assert.strictEqual(broken, 'jumped');  // Current result:  'jum'.    
```

Comment:
Ah! So you're proposing "current word" semantics! That's great! I think I know how to implement that! This is a big step up from SwiftKey which strictly considers only what's left of the cursor.

Reply:
It's something quite natural to do when a user places the caret within a word, so I figured that'd be a usability thing we'd want to see happen eventually.

Reply:
Also, I figured that's why we have LMLayer requesting text to the right-hand side of the caret for part of the `Context` spec.